scenarios/turbine-heat-processor/build.gradle

```diff
@@ -28,6 +28,7 @@ dependencies {
     compile "io.pravega:pravega-connectors-flink-${flinkMajorMinorVersion}_${flinkScalaVersion}:${flinkConnectorVersion}"
     compile "org.apache.flink:flink-streaming-java_${flinkScalaVersion}:${flinkVersion}"
     compile "org.apache.flink:flink-streaming-scala_${flinkScalaVersion}:${flinkVersion}"
+    compile "org.apache.flink:flink-clients_${flinkScalaVersion}:${flinkVersion}"
```

Comment:
The use of the `flinkScalaVersion` is confusing here. For example, we are using `flinkScalaVersion` even for the `org.apache.flink:flink-streaming-java` dependency.

Reply:
This is because the artifact name is actually `flink-streaming-java_2.12` instead of `flink-streaming-java` here. (ref: https://mvnrepository.com/artifact/org.apache.flink/flink-streaming-java_2.12/1.11.2). That's why we have to import it like this.
The reason for having such Scala version suffix for `flink-streaming-java` in Flink is because although the Java implementation is java only, the internals have some dependencies such as `flink-runtime_2.12` which contains some Scala code, so it requires the scala version to be compiled in Flink correctly.

Reply:
I'm not sure I follow why the java artifact version needs to be the same as the scala artifact version. I understand that there are internal dependencies, but the version of the java artifact should follow the `flinkJavaVersion` rather than the `flinkScalaVersion` even if they are always the same, for clarity.

We are not touching that line in this PR, so I'll request a change to that, but we should consider fixing it.

Reply:
This `2.12` is never a java version, it's the scala binary version, and should be aligned the same for all the Flink dependencies which uses Scala in the project, so it is correct to use `flinkScalaVersion`. You can also have a look at ververica usages for their projects: https://github.com/ververica/flink-cdc-connectors/blob/116cea690db4fc78814792c2383814f7e237e114/pom.xml#L106